app/main.py

```diff
@@ -1 +1,39 @@
-# write your code here
+from __future__ import annotations
+
+
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return f"{{Name: {self.name}, Health: {self.health}, Hidden: {False}}}"
+
+    def __sub__(self, other: Animal) -> Animal:
+        self.health -= other.health
+        return Animal(self.name, self.health, self.hidden)
+
+
+class Herbivore(Animal):
+    def hide(self) -> bool:
+        if self.hidden is False:
+            self.hidden = True
+        else:
+            self.hidden = False
+        return self.hidden
```

Comment:
No need to use return here, the function just change the status of hidder attribute, also u can write it in one line like:
self.hidden = not self.hidden